src/main/java/com/github/jnidzwetzki/bitfinex/v2/WebsocketClientEndpoint.java

```diff
@@ -117,37 +115,17 @@ public void onError(final Session session, final Throwable t) {
 	 * @param message
 	 */
 	public void sendMessage(final String message) {
-		
 		if(userSession == null) {
 			logger.error("Unable to send message, user session is null");
 			return;
 		}
-		
 		if(userSession.getAsyncRemote() == null) {
 			logger.error("Unable to send message, async remote is null");
 			return;
 		}
-		
 		userSession.getAsyncRemote().sendText(message);
 	}
 
-	/**
-	 * Add a new connection data consumer
-	 * @param consumer
-	 */
-	public void addConsumer(final Consumer<String> consumer) {
```

Comment:
What's wrong with the idea of registering additional low-level consumers directly on the web socket? I was thinking that this might be useful for testing.

Reply:
Testing can be done directly on component level now. Good tests cover only one layer - unless we are talking integration tests - for that we might need separate demo account (or - maybe - a dump of json communication between user and exchange?

Reply:
This is what I meant. I used the low-level callback for dumping the exchange -> client communication into a logfile. This was used in the first versions of the lib and I haven't used this callback for a long time. I think its fine that this is removed.